components/sidebar_right/sidebar_right.jsx

```diff
@@ -249,7 +196,15 @@ export default class SidebarRight extends React.PureComponent {
                     className='sidebar--right__bg'
                 />
                 <div className='sidebar-right-container'>
-                    {content}
+                    <Search
```

Comment:
What's the reason for wrapping `content` in the new `Search` component now? We don't necessarily need to change it, but it seems like this might have scaling problems if we were to change the other sidebar components to wrap each other like this.

Reply:
We did that because we wanted to have just one connected component instead of several others down the dependency tree. I understand the reasoning and I do think that you have a point (especially with the ref handling), but this should be handled in another issue (refactoring `SuggestionBox` #15855 is a big step forward)

There is still the `SearchResultItem` which is also connected, but again: this should be handled in a different issue

Reply:
I agree that those should be handled in the next steps, we need to first see how this and SuggestionBox will come together.

Reply:
I'm not sure I understand. I was meaning the structure changes going from something like
```javascript
let content;
if (threadVisible) {
    content = <Thread/>;
} else if (searchVisible) {
    content = <Search/>;
}

return content;
```
to
```javascript
let content;
if (threadVisible) {
    content = <Thread/>;
}

return <Search searchVisible={searchVisible}>{content}</Search>;
```
looked a bit odd to me since if we were following the new pattern for each RHS state (thread, search, plugin-overridden, etc), we'd end up with a more deeply nested set of components instead of only mounting the one that is needed at any given time.

Reply:
It is a decision born from integration issues with the `SearchBar` component (once with `SearchResults` in the `SidebarRight` and once without in `ChannelHeader`). Do you have another idea how this could be handled in y more "clever" way?